src/Particular.ThroughputCollector/AuditThroughput/AuditCommands.cs

```diff
@@ -2,17 +2,21 @@
 {
     using System.Text.Json.Nodes;
     using Contracts;
+    using NuGet.Versioning;
     using ServiceControl.Api;
     using AuditCount = Contracts.AuditCount;
 
     static class AuditCommands
     {
         // Customers are expected to run at least version 4.29 for their Audit instances
-        public static readonly Version MinAuditCountsVersion = new Version(4, 29);
+        public static readonly SemanticVersion MinAuditCountsVersion = new(4, 29, 0);
         // Want 2d audit retention so we get one complete UTC day no matter what time it is.
-        public static Func<RemoteInstanceInformation, bool> ValidRemoteInstances = r => r.SemVer?.Version >= MinAuditCountsVersion && r.Retention >= TimeSpan.FromDays(2);
+        public static Func<RemoteInstanceInformation, bool> ValidRemoteInstances = r =>
+            r.SemanticVersion != null &&
+            r.SemanticVersion >= MinAuditCountsVersion &&
+            r.Retention >= TimeSpan.FromDays(2);
 
-        public static async Task<ServiceControlEndpoint[]> GetKnownEndpoints(IEndpointsApi endpointsApi)
+        public static async Task<IEnumerable<ServiceControlEndpoint>> GetKnownEndpoints(IEndpointsApi endpointsApi)
```

Comment:
Use `IAsyncEnumerable`

Reply:
Yes, we need to refactor `ScatterGater` to pass cancelationtokens and IAsyncEnumeration